llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -915,51 +907,6 @@ void MachineCopyPropagation::ForwardCopyPropagateBlock(MachineBasicBlock &MBB) {
         //  %ecx = COPY %eax
         if (eraseIfRedundant(MI, Def, Src) || eraseIfRedundant(MI, Src, Def))
```

Comment:
Slightly OT for this review, but I think the eraseIfRedundant case could be removed.  The key thing it does which the main copy propagation doesn't is reason about super-registers updating sub-registers in known ways.  We could generalize the main pass, and I think this code would become redundant.  I *may* explore this further, or may not.